pkg/login/ldap_login_test.go

```diff
@@ -8,65 +8,41 @@ import (
 	"github.com/grafana/grafana/pkg/services/ldap"
 	"github.com/grafana/grafana/pkg/services/multildap"
 	"github.com/grafana/grafana/pkg/setting"
-	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 var errTest = errors.New("test error")
 
-func TestLDAPLogin(t *testing.T) {
-	Convey("Login using ldap", t, func() {
-		Convey("Given ldap enabled and no server configured", func() {
-			setting.LDAPEnabled = true
+func TestLoginUsingLDAP(t *testing.T) {
+	LDAPLoginScenario(t, "When LDAP enabled and no server configured", func(sc *LDAPLoginScenarioContext) {
+		setting.LDAPEnabled = true
```

Comment:
Not entirely within the scope of the PR, but we should fix this while we're cleaning up:
```suggestion
		origLDAPEnabled = setting.LDAPEnabled
		t.Cleanup(func() {
			setting.LDAPEnabled = origLDAPEnabled
		})
		setting.LDAPEnabled = true
```

Reply:
Good catch! I fixed it in https://github.com/grafana/grafana/pull/29998/commits/f88a3a43b287011a7869eb8d13515df169bc66c0 by preserving the original value within the `LDAPLoginScenario` function.